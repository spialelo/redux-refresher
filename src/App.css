.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

body {
  font-family: Helvetica, Arial, sans-serif;
  font-size: 16px;
  color: #222;
}

h1 {
  text-align: center;
}

form {
  text-align: center;
}

input {
  border: none;
  border-bottom: 1px dotted black;
  margin-left: 0.5em;
  font-size: 1em;
  outline: none;
}

.wrapper {
  max-width: 680px;
  margin: 0 auto;
}

.temp-wrapper {
  text-align: center;
  position: relative;
}

.temp {
  font-size: 64px;
}

.temp-symbol {
  position: absolute;
  top: 13px;
  margin-left: 4px;
}

.temp-date {
  position: absolute;
  bottom: -1em;
  left: 50%;
  transform: translateX(-50%);
  color: rgba(0, 0, 0, 0.5);
}
