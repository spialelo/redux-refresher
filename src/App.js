import React from 'react';
import './App.css';
import { connect } from 'react-redux';
import Plot from 'react-plotly.js';
import { changeLocation, setSelectedDate, setSelectedTemp, fetchData}  from './actions';


//dbe69e56e7ee5f981d76c3e77bbb45c
// ${API_KEY}

const API_KEY = process.env.REACT_APP_WEATHER_KEY;

class App extends React.Component {

  
  fetchData = (evt) => {
    evt.preventDefault();

    let location = encodeURIComponent(this.props.state.location);

    let urlPref = 'https://api.openweathermap.org/data/2.5/forecast?q=';
    let urlSuff = `&APPID=${API_KEY}&units=metric`;
    let url = urlPref + location + urlSuff;
    
    this.props.dispatch(fetchData(url));
    
  };


  onPlotClick = (data) => {
    if(data.points) {
      let number = data.points[0].pointNumber;
      this.props.dispatch(setSelectedDate(this.props.state.dates[number]));
      this.props.dispatch(setSelectedTemp(this.props.state.temps[number]));
    }
    console.log(data);
  };


  changeLocation = (evt) => {
    this.props.dispatch(changeLocation(evt.target.value));
  };

  render() {

    let currTemp = 'Not loaded yet.';
    if(this.props.state.data && this.props.state.data.list) {
      currTemp = this.props.state.data.list[0].main.temp;
    }    

      return (
        <div style={{margin: '0 auto', textAlign: 'center'}}>
          <h1>Weather v1.0</h1>
          <form onSubmit={this.fetchData}>
            <label htmlFor="usrLocation">
              I want to know the weather for
              <input 
                placeholder={"City, Country"}
                type="text"
                id="usrLocation"
                value={this.props.state.location}
                onChange={this.changeLocation}
              />
            </label>
          </form>
          {(this.props.state.data && this.props.state.data.list) ? (
            <div className="wrapper">
              { (this.props.state.selected && this.props.state.selected.temp) ? 
              <p className="temp-wrapper">
                <span className="temp-date">
                <p>The temperatures on { (this.props.state.selected && this.props.state.selected.temp) ? this.props.state.selected.date : '' }</p>
                </span>
              </p>
              :
              <p className="temp-wrapper">
                <p>The current temperature is <span className="temp">{ (this.props.state.selected && this.props.state.selected.temp) ? this.props.state.selected.temp : currTemp }
                <span className="temp-symbol">Â°C!</span></span>
                </p>
              </p>
              }
            <h2>Forecast</h2>
              <Plot 
              data={[
                {
                  x: this.props.state.dates,
                  y: this.props.state.temps,
                  type: 'scatter',
                }
              ]}
              layout={{margin: {
                    t: 0, r: 0, l: 30
                },
                xaxis: {
                    gridcolor: 'transparent'
                }}}
              displayModeBar={false}
              onClick={this.onPlotClick}
            />
          </div>
          )
          : null}
          
        </div>
      );
    }
}

function mapStateToProps(state) {
  return {
    state
  }
}

export default connect(mapStateToProps)(App);
