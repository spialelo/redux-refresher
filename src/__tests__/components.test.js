import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { App } from '../App';
import Plot from 'react-plotly.js';

// How to test React components developed using React-Plotly
// https://github.com/plotly/react-plotly.js/issues/176

describe.skip('components', function() {
    describe('<App />', function() {
        it('renders correctly', function() {
            var tree = renderer.create(<App redux={fromJS({})} />).toJSON();
          expect(tree).toMatchSnapshot();
        });
    });
});

