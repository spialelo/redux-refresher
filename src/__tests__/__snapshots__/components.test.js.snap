// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <App /> renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "textAlign": "center",
    }
  }
>
  <h1>
    Weather v1.0
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="usrLocation"
    >
      I want to know the weather for
      <input
        id="usrLocation"
        onChange={[Function]}
        placeholder="City, Country"
        type="text"
      />
    </label>
  </form>
</div>
`;
